### PR TITLE
Builder-Vite: Add null character prefix to virtual file IDs in code generator plugin

### PR DESCRIPTION
**Pull request by valentinpalkovic**
Closes #

## What I did

Cleaned up the code for resolving and loading virtual files in the Vite builder.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

## Greptile Summary

This PR modifies the virtual file handling in the Vite builder for Storybook, improving compatibility with Vite's virtual module system.

- Added `resolvedVirtualPath` function in `code-generator-plugin.ts` to prefix virtual file IDs with a null byte
- Updated virtual file ID resolution throughout the plugin to use the new `resolvedVirtualPath` function
- Removed leading forward slashes from virtual file paths in `virtual-file-names.ts`
- These changes ensure better alignment with Vite's expected format for virtual modules

### CODE REVIEW

File: code/builders/builder-vite/src/virtual-file-names.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export const virtualFileId = '/virtual:/@storybook/builder-vite/vite-app.js';
```
Comment (valentinpalkovic) [inline code/builders/builder-vite/src/virtual-file-names.ts:1]: @IanVS Is the leading forward slash required?

**Comment by IanVS** [comment]
Thanks for doing this, but it looks like I was wrong.  I found a guide to virtual modules which I'm pretty sure must be new, or at least I don't remember seeing it before, but it's very helpful: https://vitejs.dev/guide/api-plugin.html#virtual-modules-convention.  Looks like you did it the right way, though I guess maybe a slightly cleaner approach is to create another constant for the ID used in resolving.

File: code/builders/builder-vite/src/virtual-file-names.ts (hunk 1 above)
Comment (IanVS) [reply code/builders/builder-vite/src/virtual-file-names.ts:1]: Not as far as I know.

**Comment by valentinpalkovic** [comment]
@IanVS Can you take another look?

**Review by IanVS — APPROVED** [review]
Seems like a nice cleanup.

**Comment by valentinpalkovic** [comment]
Superseded by https://github.com/storybookjs/storybook/pull/28984

**Pull request closed without merging**